analysis/freq.go

```diff
@@ -123,13 +36,13 @@ func TokenFrequency(tokens TokenStream, arrayPositions []uint64, includeTermVect
 			curr, ok := rv[string(token.Term)]
 			if ok {
 				curr.Locations = append(curr.Locations, &tls[tlNext])
-				curr.frequency++
+				curr.SetFrequency(curr.Frequency() + 1)
 			} else {
-				rv[string(token.Term)] = &TokenFreq{
+				rv[string(token.Term)] = &index.TokenFreq{
 					Term:      token.Term,
-					Locations: []*TokenLocation{&tls[tlNext]},
-					frequency: 1,
+					Locations: []*index.TokenLocation{&tls[tlNext]},
 				}
+				rv[string(token.Term)].SetFrequency(1)
```

Comment:
As rv looks like map[string]* and  there is string(byte) conversion during look ups,
can we just access it only once for setting it like on the original code with a fully initialized &index.TokenFreq{}. (saves an extra map look up and the string-[]byte conversion).

Reply:
I don't think the string is actually converted, because of this optimization in the runtime: https://github.com/golang/go/commit/f5f5a8b6209f84961687d993b93ea0d397f5d5bf

But I'll rewrite and see if you like that better.